File: webapp/src/Client/Person.ts
import { useCRUD, UseCRUDOptionalProps } from "./CRUD/CRUD";
import { RoleAssignment } from "./RoleAssignment";

export interface Person {
  id: number;
  external_id: string;
  first_name: string;
  middle_names?: string;
  last_name: string;
  active_role_assignment: RoleAssignment;
}

export interface MutatePerson {
  external_id: string;
  first_name: string;
  middle_names?: string;
  last_name: string;
}

export interface UsePersonProps extends UseCRUDOptionalProps { }

export function usePerson(props?: UsePersonProps) {
  const {
    reload,
    items,
    loading,
    error,
    createItem,
    retrieveItem,
    updateItem,
    deleteItem,
  } = useCRUD<Person, MutatePerson>({
    path: "/api/org/persons",
    ...props,
  });
  return {
    items,
    loading,
    error,
    reload,
    createItem,
    retrieveItem,
    updateItem,
    deleteItem,
  };
}

File: cs/org/src/Entities/Person.cs
using Base.Entities;

namespace Org.Entities;

public class Person : EntityBase
{
  public string ExternalId { get; set; } = null!;

  public string FirstName { get; set; } = null!;
  public string? MiddleNames { get; set; }
  public string LastName { get; set; } = null!;

  public List<RoleAssignment> RoleAssignments { get; set; } = new();
}
